### ISSUE TITLE
Clarify shadow block rules

### ISSUE THREAD

**Issue by RoboErikG** [google/blockly#304]
**Can shadow blocks contain other blocks?**
- Just other shadows or non-shadow blocks as well?
- If a shadow block contains a shadow block can the inner block be replaced by a normal block?
- If a shadow block contains a normal block what happens when you replace the shadow block?
  - Does the normal block get hidden along with the shadow it's attached to?
  - Does the normal block get bumped out?

Answered questions:
**Can a shadow block be a top level block?**
No promise to support this use case, but it is not explicitly disallowed.

**Comment by RoboErikG** [comment]
Note: currently Blockly is perfectly happy to have blocks within shadow blocks and it just hides the whole stack when you replace the shadow block.

**Comment by RoboErikG** [comment]
Comments copied from the Android repo https://github.com/google/blockly-android/issues/199 so we can track completion of this on both platforms.

After some internal discussion we've decided to support complex shadow blocks in the following ways on all platforms.

- Disallow top level shadow blocks.
- Allow complex shadow blocks, but only allow shadows to have other shadows as children (enforced at xml load time and while dragging).
- Bump attempts to connect a normal block to a shadow block.
- Strongly suggest developers only use simple* shadows for kid focused apps.

*A simple block is a block with only an output or a previous connection (no next or value connections).

@NeilFraser One more detail--when connecting a block with a shadow next between two other blocks I opted to bump the remainder in Android instead of replacing the shadow block on next.

Does that seem reasonable to you?
